common-tasks/usb.md

```diff
@@ -383,6 +293,185 @@ When you finish, detach the device:
+-------------------
+
+**Warning:** This procedure will result in your USB controller(s) being attached
+directly to dom0.
+
+1. Shut down the USB qube.
+2. In Qubes Manager, right-click on the USB qube and select "Remove VM."
+3. Open the file `/etc/default/grub` in dom0.
+4. Find the line(s) that begins with `GRUB_CMDLINE_LINUX`.
+5. If `rd.qubes.hide_all_usb` appears anywhere in those lines, remove it.
+6. Save and close the file.
+7. Run the command `grub2-mkconfig -o /boot/grub2/grub.cfg` in dom0.
+8. Reboot.
+
+
+Security Warning about USB Input Devices
+----------------------------------------
+
+**Important security warning. Please read this section carefully!**
+
+If you connect USB input devices (keyboard and mouse) to a VM, that VM will effectively have control over your system.
+Because of this, the benefits of using a USB qube are much smaller than using a fully untrusted USB qube.
+In addition to having control over your system, such VM can also sniff all the input you enter there (for example, passwords in the case of a USB keyboard).
+
+There is no simple way to protect against sniffing, but you can make it harder to exploit control over input devices.
+
+If you have only a USB mouse connected to a USB qube, but the keyboard is connected directly to dom0 (using a PS/2 connector, for example), you simply need to lock the screen when you are away from your computer.
+You must do this every time you leave your computer unattended, even if there no risk of anyone else having direct physical access to your computer.
+This is because you are guarding the system not only against anyone with local access, but also against possible actions from a potentially compromised USB qube.
+
+If your keyboard is also connected to a USB qube, things are much harder.
+Locking the screen (with a traditional password) does not solve the problem, because the USB qube can simply sniff this password and later easily unlock the screen.
+One possibility is to set up the screen locker to require an additional step to unlock (i.e., two-factor authentication).
+One way to achieve this is to use a [YubiKey], or some other hardware token, or even to manually enter a one-time password.
+
+How to use a USB keyboard
+-------------------------
+
+**Caution:** Please carefully read the [Security Warning about USB Input Devices] before proceeding.
+
+In order to use a USB keyboard, you must first attach it to a USB qube, then give that qube permission to pass keyboard input to dom0.
+Edit the `qubes.InputKeyboard` policy file in dom0, which is located here:
+
+    /etc/qubes-rpc/policy/qubes.InputKeyboard
+
+Add a line like this one to the top of the file:
+
+    sys-usb dom0 allow,user=root
+
+(Change `sys-usb` to your desired USB qube.)
+
+You can now use your USB keyboard.
+
+For a confirmation dialog each time the USB keyboard is connected, change this line to:
+```
+sys-usb dom0 ask,default_target=dom0
+```
+
+How to use a USB mouse
+----------------------
```

Comment:
Maybe worth noting that this is also done as part of `qubesctl state.sls qvm.sys-usb` (or if user selected USB qube during first boot).